fundingmanager.go

```diff
@@ -840,109 +838,125 @@ func (f *fundingManager) reservationCoordinator() {
 // advanceFundingState will advance the channel fromt the markedOpen state to
 // the point where the channel is ready for operation. This includes sending
 // funding locked to the peer, adding the channel to the router graph, and
-// announcing the channel.
-func (f *fundingManager) advanceFundingState(
-	channel *channeldb.OpenChannel) error {
-
-	channelState, shortChanID, err := f.getChannelOpeningState(
-		&channel.FundingOutpoint)
-	if err == ErrChannelNotFound {
-		// Channel not in fundingManager's opening database,
-		// meaning it was successfully announced to the
-		// network.
-		return nil
-	} else if err != nil {
-		return err
+// announcing the channel. The updateChan can be set non-nil to get
+// OpenStatusUpdates.
+//
+// NOTE: This MUST be run as a goroutine.
+func (f *fundingManager) advanceFundingState(channel *channeldb.OpenChannel,
+	updateChan chan<- *lnrpc.OpenStatusUpdate) {
+	defer f.wg.Done()
+
+	// We create the state-machine object which wraps the database state.
+	lnChannel, err := lnwallet.NewLightningChannel(
+		nil, channel, nil,
+	)
+	if err != nil {
+		fndgLog.Errorf("Unable to create LightningChannel(%v): %v",
+			channel.FundingOutpoint, err)
+		return
+	}
+
+	for {
+		channelState, shortChanID, err := f.getChannelOpeningState(
```

Comment:
Do we need to read it from disk each time? We already know what the new state is, and only this loop can modify the the state.

Reply:
It is to simplify the state machine, so it can easily be resumed from any state. Returning the new state from `stateStep` to only read it once would be an optimization, but now sure how critical.

Reply:
Would also need to return the new state from the `stateStep` method, and define a new state or error for when to exit the loop.

Reply:
Fine with me either way, doesn't seem very critical.